include/zephyr/net/net_if.h

```diff
@@ -2643,6 +2643,35 @@ void net_if_unregister_link_cb(struct net_if_link_cb *link);
 void net_if_call_link_cb(struct net_if *iface, struct net_linkaddr *lladdr,
 			 int status);
 
+/* used to ensure encoding of checksum support in net_if.h and
+ * ethernet.h is the same
+ */
+#define NET_IF_CHECKSUM_NONE_BIT			0
+#define NET_IF_CHECKSUM_IPV4_HEADER_BIT			BIT(0)
+#define NET_IF_CHECKSUM_IPV4_ICMP_BIT			BIT(1)
+#define NET_IF_CHECKSUM_IPV6_HEADER_BIT			BIT(10)
+#define NET_IF_CHECKSUM_IPV6_ICMP_BIT			BIT(11)
+#define NET_IF_CHECKSUM_TCP_BIT				BIT(21)
+#define NET_IF_CHECKSUM_UDP_BIT				BIT(22)
+
+/**
+ * @brief Type of checksum for which support in the interface will be queried.
+ */
+enum net_if_checksum_type {
+	NET_IF_CHECKSUM_IPV4_HEADER = NET_IF_CHECKSUM_IPV4_HEADER_BIT,
+	NET_IF_CHECKSUM_IPV4_TCP = NET_IF_CHECKSUM_IPV4_HEADER_BIT|
```

Comment:
I'm not sure I follow, why check for IPV4_HEADER here? I'm not sure what the actual hardware support in the matter, but those doesn't seem related, L4/L3 checksum calculation should IMO be independent.

Reply:
This is more important for IPv6 - if a network device only supports IPv4 and TCP checksums (in IPv4 packets), then we have to compute IPv6 checksums in software.
The check for IPv4 support could theoretically be omitted (I am not aware of any devices that support IPv6 checksums but do not support IPv4).

Reply:
I am not sure if this answers your question, but L3 support is required for L4 support. A NIC that does not understand the L3 protocol has no idea where the L4 packet begins and ends, what transport protocol it is and where its checksum field would be. Thus, it cannot insert an L4 checksum.

Reply:
Ok, I see your point, fine by me then.